src/Core/AndroidNative/maui/src/main/java/com/microsoft/maui/ResizeTransformation.java

```diff
@@ -0,0 +1,57 @@
+package com.microsoft.maui;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.util.DisplayMetrics;
+import com.bumptech.glide.load.engine.bitmap_recycle.BitmapPool;
+import com.bumptech.glide.load.resource.bitmap.BitmapTransformation;
+import com.bumptech.glide.load.Key;
+import java.nio.ByteBuffer;
+import java.security.MessageDigest;
+
+class ResizeTransformation extends BitmapTransformation {
+    private static final String ID = "com.microsoft.maui.ResizeTransformation";
+    private static final byte[] ID_BYTES = ID.getBytes(Key.CHARSET);
+    private DisplayMetrics displayMetrics;
+
+    public ResizeTransformation(DisplayMetrics display) {
+        this.displayMetrics = display;
+    }
+
+    @Override
+    protected Bitmap transform(BitmapPool pool, Bitmap toTransform, int outWidth, int outHeight) {
+        int width = toTransform.getWidth();
+        int height = toTransform.getHeight();
+
+        if (width <= displayMetrics.widthPixels && height <= displayMetrics.heightPixels) {
+            return toTransform;
+        }
+
+        float aspectRatio = (float) width / (float) height;
+
+        if (width > height) {
+            outWidth = displayMetrics.widthPixels;
+            outHeight = Math.round(displayMetrics.widthPixels / aspectRatio);
+        } else {
+            outHeight = displayMetrics.heightPixels;
+            outWidth = Math.round(displayMetrics.heightPixels * aspectRatio);
+        }
+
+        return Bitmap.createScaledBitmap(toTransform, outWidth, outHeight, false);
+    }
+
+    @Override
+     public boolean equals(Object o) {
+       return o instanceof ResizeTransformation;
+     }
+
+      @Override
+     public int hashCode() {
+       return ID.hashCode();
+     }
+
+      @Override
+     public void updateDiskCacheKey(MessageDigest messageDigest) {
+       messageDigest.update(ID_BYTES);
```

Comment:
Should this take into account the display width/height as part of the cache key / hashcode implementation?  In theory you could have multiple displays and the activity move between them and want a larger image than was first transformed but still within the display size?

Probably very edge case, but adding the display width/height to the hashcode and cache key wouldn't negatively impact the typical case either, so maybe worth doing to be safe?

Reply:
I'll see if I can implement that. If I'm honest I wasn't too sure what the `MessageDigest` was for. It just appears in allot of code 🤣